connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -231,6 +232,8 @@ public class DistributedHerder extends AbstractHerder implements Runnable {
     Thread herderThread;
 
     private final DistributedConfig config;
+    
+    public static Future<?> herderFuture;
```

Comment:
I don't think this can be `static`; if it is, we won't be able to use it with multiple workers in a single integration test (which we do very frequently).

Reply:
I have Imporved this place, PTAL.